src/sourmash/search.py

```diff
@@ -393,22 +497,47 @@ def __post_init__(self):
         self.init_sigcomparison() # initialize original sketch vs match sketch comparison (inherited from PrefetchResult)
         self.init_gathersketchcomparison() # initialize remaining gather sketch vs match sketch comparison
         self.build_gather_result() # build gather-specific attributes
+        # set write columns for prefetch result
+        self.write_cols = self.gather_write_cols
+        if self.estimate_ani_ci:
+            self.write_cols = self.gather_write_cols_ci
 
-    def gatherresultdict(self):
+    def prep_gather_result(self):
         # for gather, we only shorten the query_md5
+        self.scaled = self.cmp_scaled
         self.query_md5 = self.shorten_md5(self.query_md5)
-        return self.to_write(columns=self.gather_write_cols)
+
+    def prep_result(self):
+        # overwrite base prep_result
+        self.prep_gather_result()
 
     @property
-    def writedict(self):
-        return self.gatherresultdict()
+    def gatherresultdict(self):
+        # just return dictionary of what we want to write
+        self.prep_gather_result()
+        return self.to_write(columns=self.write_cols)
 
     @property
-    def prefetchwritedict(self):
-        # enable writing prefetch csv from a GatherResult
-        self.build_prefetch_result()
-        return self.prefetchresultdict()
+    def prefetchresultdict(self):
+        # get all prefetch cols from gatherresult
+        prefetch_cols = self.prefetch_write_cols
+        if self.estimate_ani_ci:
+            prefetch_cols = self.prefetch_write_cols_ci
+        self.jaccard = self.cmp.jaccard
+        self.f_query_match = self.cmp.mh2_containment #db_mh.contained_by(query_mh)
+        self.f_match_query = self.cmp.mh1_containment #query_mh.contained_by(db_mh)
+        self.prep_prefetch_result()
+        return self.to_write(columns=prefetch_cols)
+
+    def init_prefetch_dictwriter(self, csv_handle):
```

Comment:
Thinking of deleting this function. My thinking here was that we could print prefetch results directly from gather results, but I think it will always be best to generate prefetch results first, so we probably don't need this.

@ctb  thoughts?

Reply:
I agree it's probably not needed!